include/keyboard.js

```diff
@@ -285,6 +285,134 @@ var kbdUtil = (function() {
     };
 })();
 
+function QEMUKeyEventDecoder(modifierState, next) {
+    "use strict";
+    function sendAll(evts) {
+        for (var i = 0; i < evts.length; ++i) {
+            next(evts[i]);
+        }
+    }
+
+    function isNumPadMultiKey(evt) {
+        var numPadCodes = ["Numpad0", "Numpad1", "Numpad2",
+            "Numpad3", "Numpad4", "Numpad5", "Numpad6",
+            "Numpad7", "Numpad8", "Numpad9", "NumpadDecimal"];
+        return (numPadCodes.indexOf(evt.code) !== -1);
+    }
+
+    function getNumPadKeySym(evt) {
+        var numLockOnKeySyms = {
+            "Numpad0": 0xffb0, "Numpad1": 0xffb1, "Numpad2": 0xffb2,
```

Comment:
these maps and lists should probably be lifted outside of the functions or turned into switch statements.